test/fixtures/03-async-migrator/04-migration-errors/migrations/0001-successful-migration.async.ts

```diff
@@ -0,0 +1,35 @@
+import { AsyncMigration } from '../../../../../src/migrator/utils';
+
+export = {
+	asyncFn: async (tx: any, options) => {
+		const staticSql = `\
+		UPDATE "device"
+		SET "note" = "device"."name"
+		WHERE id IN (   SELECT id FROM (
+							SELECT id FROM "device"
+							WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL
+							LIMIT ${options.batchSize}
+							) tmp
+					);
+        `;
+
+		return await tx.executeSql(staticSql);
+	},
+	syncFn: async (tx: any) => {
+		const staticSql = `\
+		UPDATE "device"
+		SET "note" = "device"."name"
+		WHERE id IN (   SELECT id FROM (
+							SELECT id FROM "device"
+							WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL
+							) tmp
+					);
+        `;
+
+		await tx.executeSql(staticSql);
+	},
+	asyncBatchSize: 1,
+	delayMS: 250,
+	backoffDelayMS: 4000,
+	errorThreshold: 15,
+} as AsyncMigration;
```

Comment:
We should be able to get type inference & better type checking if we change this (and all other tests) to:
```suggestion
const migrations: AsyncMigration = {
	asyncFn: async (tx, options) => {
		const staticSql = `\
		UPDATE "device"
		SET "note" = "device"."name"
		WHERE id IN (   SELECT id FROM (
							SELECT id FROM "device"
							WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL
							LIMIT ${options.batchSize}
							) tmp
					);
        `;

		return await tx.executeSql(staticSql);
	},
	syncFn: async (tx) => {
		const staticSql = `\
		UPDATE "device"
		SET "note" = "device"."name"
		WHERE id IN (   SELECT id FROM (
							SELECT id FROM "device"
							WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL
							) tmp
					);
        `;

		await tx.executeSql(staticSql);
	},
	asyncBatchSize: 1,
	delayMS: 250,
	backoffDelayMS: 4000,
	errorThreshold: 15,
};
export = migration;
```